pkg/types/openstack/failuredomain.go

```diff
@@ -0,0 +1,6 @@
+package openstack
+
+import apiconfig "github.com/openshift/api/config/v1"
+
+// FailureDomain holds the information for a failure domain
+type FailureDomain = apiconfig.OpenStackFailureDomain
```

Comment:
Could we also have `bgpConfiguration` defined (in its own type of course), so in the install-config we can configure the BGP configuration, whether it's embedded in the `FailureDomain` or not is another [discussion](https://github.com/openshift/api/pull/1243/files#r965175219).

Reply:
Maintaining an open PR for a long time is work. I'd suggest to identify the minimum set of changed this PR needs before merging, and then merging it before even starting to integrate any notion of BGP into the Installer.

Note that what I'm proposing also entails merging [the api PR](https://github.com/openshift/api/pull/1243) with the FailureDomain bits only.